ProjFS.Mac/PrjFSLib/PrjFSLib.cpp

```diff
@@ -1042,29 +1057,35 @@ static PrjFS_Result HandleNewFileInRootNotification(
     const MessageHeader* request,
     const char* relativePath,
     const char* absolutePath,
+    const char* relativeFromPath,
     bool isDirectory,
     PrjFS_NotificationType notificationType)
 {
 #ifdef DEBUG
     cout
         << "HandleNewFileInRootNotification: "
-        << absolutePath
-        << " (root-relative: " << relativePath << ")"
+        << (absolutePath == nullptr ? "[NULL]" : absolutePath)
+        << " (root-relative: " << (relativePath == nullptr ? "[NULL]" : relativePath) << ")"
+        << " (root-relative from: " << (relativeFromPath == nullptr ? "[NULL]" : relativeFromPath) << ")"
         << " Process name: " << request->procname
         << " Pid: " << request->pid
         << " notificationType: " << NotificationTypeToString(notificationType)
         << " isDirectory: " << isDirectory << endl;
 #endif
 
-    // Whenever a new file shows up in the root, we need to check if its ancestor
-    // directories are flagged as in root.  If they are not, flag them as in root and
-    // notify the provider
-    FindNewFoldersInRootAndNotifyProvider(request, relativePath);
+    if (relativePath != nullptr)
```

Comment:
Same comment here, in `HandleNewFileInRootNotification` I would expect `relativePath` and `absolutePath` to always be non-null, and `relativeFromPath` to always be `nullptr`.

Assuming that's the case I don't think we should add code to handle `nullptr` when it's not expected.

Reply:
I see now what's going on.  See comments above when we call `HandleNewFileInRootNotification`.  We should only call `HandleNewFileInRootNotification` when the hardlink results in a new file inside the root.